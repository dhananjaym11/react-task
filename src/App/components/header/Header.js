import React from 'react';
import { Link } from "react-router-dom";

import './Header.scss';

const Header = () => (
    <header className="app-header">
        <Link to="/" className="app-logo">
            <span>Logo</span>
        </Link>

        <div className="float-right">
            <ul className="header-links clearfix">
                <li>
                    <Link to="/students">
                        <span>Students</span>
                    </Link>
                </li>
            </ul>
        </div>
    </header>
);

export default Header;