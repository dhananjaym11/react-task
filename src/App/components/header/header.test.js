import React from 'react';
import { shallow } from 'enzyme';

import Header from './Header';

it('Test header component', () => {
    expect(shallow(<Header />)).toMatchSnapshot();
});