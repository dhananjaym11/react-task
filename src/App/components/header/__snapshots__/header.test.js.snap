// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test header component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="app-logo"
          replace={false}
          to="/"
        >
          <span>
            Logo
          </span>
        </Link>,
        <div
          className="float-right"
        >
          <ul
            className="header-links clearfix"
          >
            <li>
              <Link
                replace={false}
                to="/students"
              >
                <span>
                  Students
                </span>
              </Link>
            </li>
          </ul>
        </div>,
      ],
      "className": "app-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <span>
            Logo
          </span>,
          "className": "app-logo",
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Logo",
          },
          "ref": null,
          "rendered": "Logo",
          "type": "span",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul
            className="header-links clearfix"
          >
            <li>
              <Link
                replace={false}
                to="/students"
              >
                <span>
                  Students
                </span>
              </Link>
            </li>
          </ul>,
          "className": "float-right",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <li>
              <Link
                replace={false}
                to="/students"
              >
                <span>
                  Students
                </span>
              </Link>
            </li>,
            "className": "header-links clearfix",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/students"
              >
                <span>
                  Students
                </span>
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <span>
                  Students
                </span>,
                "replace": false,
                "to": "/students",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Students",
                },
                "ref": null,
                "rendered": "Students",
                "type": "span",
              },
              "type": [Function],
            },
            "type": "li",
          },
          "type": "ul",
        },
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="app-logo"
            replace={false}
            to="/"
          >
            <span>
              Logo
            </span>
          </Link>,
          <div
            className="float-right"
          >
            <ul
              className="header-links clearfix"
            >
              <li>
                <Link
                  replace={false}
                  to="/students"
                >
                  <span>
                    Students
                  </span>
                </Link>
              </li>
            </ul>
          </div>,
        ],
        "className": "app-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <span>
              Logo
            </span>,
            "className": "app-logo",
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Logo",
            },
            "ref": null,
            "rendered": "Logo",
            "type": "span",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <ul
              className="header-links clearfix"
            >
              <li>
                <Link
                  replace={false}
                  to="/students"
                >
                  <span>
                    Students
                  </span>
                </Link>
              </li>
            </ul>,
            "className": "float-right",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <li>
                <Link
                  replace={false}
                  to="/students"
                >
                  <span>
                    Students
                  </span>
                </Link>
              </li>,
              "className": "header-links clearfix",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/students"
                >
                  <span>
                    Students
                  </span>
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <span>
                    Students
                  </span>,
                  "replace": false,
                  "to": "/students",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Students",
                  },
                  "ref": null,
                  "rendered": "Students",
                  "type": "span",
                },
                "type": [Function],
              },
              "type": "li",
            },
            "type": "ul",
          },
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
